### ISSUE TITLE
CUDA out of memory

### ISSUE THREAD

**Issue by ZZZsleepyheadZZZ** [alexriggio/BERT-LoRA-TensorRT#4]
Thank you for the well-organized code and detailed notes, which makes the life more easier for a beginner like me.

However during my evaluating process I came up with the 'CUDA out of memory' error (when the training process is done, and. there is no other running process). I doubted that it may be caused by memory leak issue.

**Comment by alexriggio** [comment]
Yeah that is strange there is an issue with just evaluating considering fine-tuning also includes an evaluation step for each epoch.  

Some quick things to try:

Just to confirm, when you restart the kernel (clearing the memory) and go straight to evaluating (skip over the fine-tuning) you still run into the 'CUDA out of memory' error? 

Another workaround is to try reducing the batch size. In `STEP 1: Data Preprocessing`, set `batch size=1` for the `test_dataloader`.
